R/tm_g_patient_profile.R

```diff
@@ -509,6 +503,65 @@ srv_g_patient_profile <- function(id,
 
     # render plot
     output_q <- reactive({
+      iv <- shinyvalidate::InputValidator$new()
```

Comment:
I don't think the validation as is works here - as depending on what arguments the app developer gives some of the ui isn't there:
![image](https://user-images.githubusercontent.com/15201933/206425812-662d0e7f-7a19-4df9-a549-797ab3984b18.png)

From:

```
library(scda)
library(nestcolor)

latest_data <- synthetic_cdisc_data("latest")

ADSL <- latest_data$adsl
ADAE <- latest_data$adae %>%
  mutate(
    ASTDT = as.Date(ASTDTM),
    AENDT = as.Date(AENDTM)
  )


ADRS <- latest_data$adrs %>%
  mutate(ADT = as.Date(ADTM))


x <- init(
  data = cdisc_data(
    cdisc_dataset("ADSL", ADSL,
      code = "ADSL <- synthetic_cdisc_data(\"latest\")$adsl"
    ),
    cdisc_dataset("ADRS", ADRS,
      code = "ADRS <- synthetic_cdisc_data(\"latest\")$adrs %>%
              mutate(ADT = as.Date(ADTM))"
    ),
    cdisc_dataset("ADAE", ADAE,
      code = "ADAE <- synthetic_cdisc_data(\"latest\")$adae %>%
              mutate(ASTDT = as.Date(ASTDTM),
                     AENDT = as.Date(AENDTM))"
    ),
    check = FALSE # set FALSE here to keep run time of example short, should be set to TRUE
  ),
  modules = modules(
    tm_g_patient_profile(
      label = "Patient Profile Plot",
      patient_id = choices_selected(
        choices = unique(ADSL$USUBJID),
        selected = unique(ADSL$USUBJID)[1]
      ),
      sl_dataname = "ADSL",
      ae_dataname = "ADAE",
      rs_dataname = "ADRS",
      lb_dataname = NA_character_,
      ex_dataname = NA_character_,
      cm_dataname = NA_character_,
      show_ex_plot = FALSE,
      show_ae_plot = TRUE,
      show_rs_plot = TRUE,
      show_cm_plot = FALSE,
      show_lb_plot = TRUE,
      sl_start_date = choices_selected(
        selected = "TRTSDTM",
        choices = c("TRTSDTM", "RANDDT")
      ),
      ex_var = choices_selected(
        selected = "PARCAT2",
        choices = "PARCAT2"
      ),
      ae_var = choices_selected(
        selected = "AEDECOD",
        choices = c("AEDECOD", "AESOC")
      ),
      ae_line_col_var = choices_selected(
        selected = "AESER",
        choices = c("AESER", "AEREL")
      ),
      ae_line_col_opt = c("Y" = "red", "N" = "blue"),
      rs_var = choices_selected(
        selected = "PARAMCD",
        choices = "PARAMCD"
      ),
      cm_var = choices_selected(
        selected = "CMDECOD",
        choices = c("CMDECOD", "CMCAT")
      ),
      lb_var = choices_selected(
        selected = "LBTESTCD",
        choices = c("LBTESTCD", "LBCAT")
      ),
      x_limit = "-28, 750",
      plot_height = c(1200, 400, 5000)
    )
  )
)
if (interactive()) {
  shinyApp(x$ui, x$server)
}
```